### ISSUE TITLE
Pidgin: horde of windows open on startup

### ISSUE THREAD

**Issue by ildar** [matrix-org/purple-matrix#75]
On Pidgin start up I get all the rooms open. Most of them have no new messages. That's very inconvenient.

**Comment by EionRobb** [comment]
Try https://github.com/kgraefe/pidgin-hide-chat-on-join

**Comment by teleshoes** [comment]
@EionRobb hide-chat-on-join indiscriminately hides any chat you tell it to, regardless of whether you have unread messages at startup, and is therefore more or less useless

ideally, purple-matrix should just not open empty chats at startup; if its empty, dont show it

**Comment by grimmy** [comment]
Sure matrix could not show the chat, but then it would have to add all of them to the buddy list as that's how this behavior worked on past chat protocols.  As such, purple has no logic to determine whether a conversation be shown or not based on new messages.

That said, I imagine it'd be less annoying if the chat history was displayed?  If that's the case, there are plugins that'll do this, but it'll only displays history that purple already saw, not messages that were received as the libpurple based client was offline.

Until there are some better interfaces in purple, that's the only option.  That said, any new interfaces like this are going in purple3 and not purple2 and due to our skeleton crew right now purple3 is still quite a ways out before even an alpha release.  Likewise, due to the lack of contributors all new developments to purple2 mostly come from pull requests as the core devs have to stay focused to fix the fragmentation.

**Comment by teleshoes** [comment]
it is not less annoying with history. i have 36 pidgin tabs at startup. when i receive a message, i have to do ctrl+tab to see who its from, or horizontally scroll through tabs like a peasant.

i understand that there is no in-built api in purple for this, but this feature-gap makes pidgin more or less useless as a matrix client.

(at startup, some tabs are blue with new messages, and most tabs are not. if a plugin like hide-chat could be written to close all tabs except those, it would fill that feature-gap)

**Comment by EionRobb** [comment]
There's a "chatpop" plugin in https://developer.pidgin.im/ticket/5507 that might work well with the hide-on-join plugin to help workaround this?

**Comment by teleshoes** [comment]
WOOO!! it works perfectly, when combined with hide-on-join. thank you SO much! @EionRobb :D

**Comment by teleshoes** [comment]
had to tweak it, as it focused the window on every message
https://github.com/teleshoes/chatpop-pidgin

**Comment by g4borg** [comment]
yes the combination of those two plugins does do the trick, thank you for the modded version.
however it would be still awesome to have this functionality natively (remember the last open chats, only open them if there are new messages, do not open ALL of them each time)

**Comment by axet** [comment]
Direct approach isnt' working here, you can see it in (gtkconv.c) :

```cpp
static void
hide_conv(PidginConversation *gtkconv, gboolean closetimer)
```

To hide coverstaion window, you need to call hide_conv, which is 'static' and you need PidginConversation, which is different from PurpleConversation.
